apps/users/emails.py

```diff
@@ -46,10 +48,24 @@ def get_attachments(self):
 
         organisation = self.get_organisation()
         if organisation and organisation.logo:
-            f = open(organisation.logo.path, "rb")
-            logo = MIMEImage(f.read())
-            logo.add_header("Content-ID", "<{}>".format("organisation_logo"))
-            attachments += [logo]
+            with open(organisation.logo.path, "rb") as f:
+                logo = None
+                data = f.read()
+                try:
+                    logo = MIMEImage(data)
+                except TypeError:
+                    capture_message(
+                        "warning: MIMEImage failed to detect mime type:\n"
+                        "organisation:"
+                        + organisation.name
+                        + "\nfile:"
+                        + organisation.logo.path
+                    )
+                    mime_type = magic.from_buffer(data, mime=True)
+                    logo = MIMEImage(data, _subtype=mime_type)
+                if logo:
```

Comment:
Better decouple the logo checking from the `with` block, and so the first assignment of the logo set just before the `with`.